sdk/include/opentelemetry/sdk/trace/etw_data.h

```diff
@@ -0,0 +1,530 @@
+    UNREFERENCED_PARAMETER(name);
+  }
+
+  /// <summary>
+  /// End Span
+  /// </summary>
+  /// <returns></returns>
+  virtual void End(const trace::EndSpanOptions & = {}) noexcept override
+  {
+    // TODO: signal this to owner
+  }
+
+  virtual trace::SpanContext GetContext() const noexcept
+  {
+    // TODO: not implemented
+    static trace::SpanContext nullContext{trace::SpanContext::GetInvalid()};
+    return nullContext;
+  }
+
+  /// <summary>
+  /// Check if Span is recording data
+  /// </summary>
+  /// <returns></returns>
+  virtual bool IsRecording() const noexcept override
+  {
+    // TODO: not implemented
+    return true;
+  }
+
+  virtual void SetToken(nostd::unique_ptr<context::Token> &&token) noexcept
+  {
+    // TODO: not implemented
+    UNREFERENCED_PARAMETER(token);
+  }
+
+  /// <summary>
+  /// Get Owner tracer of this Span
+  /// </summary>
+  /// <returns></returns>
+  trace::Tracer &tracer() const noexcept { return this->owner; };
+};
+
+/// <summary>
+/// stream::TraceProvider
+/// </summary>
+class TracerProvider : public trace::TracerProvider
+{
+public:
+  /// <summary>
+  /// Obtain a Tracer of given type (name) and supply extra argument arg2 to it.
+  /// </summary>
+  /// <param name="name">Tracer Type</param>
+  /// <param name="args">Tracer arguments</param>
+  /// <returns></returns>
+  virtual nostd::shared_ptr<trace::Tracer> GetTracer(nostd::string_view name,
+                                                     nostd::string_view args = "")
+  {
+    // TODO: describe possible args. Currently supported:
+    // "MSGPACK"        - use MessagePack
+    // "XML"            - use XML
+    // "ETW"            - use 'classic' Trace Logging Dynamic manifest ETW event
+    //
+#if defined(HAVE_NO_TLD) && defined(HAVE_MSGPACK)
+    ETWProvider::EventFormat evtFmt = ETWProvider::EventFormat::ETW_MSGPACK;
+#else
+    ETWProvider::EventFormat evtFmt = ETWProvider::EventFormat::ETW_MANIFEST;
+#endif
+
+#pragma warning(push)
+#pragma warning(disable : 4307) /* Integral constant overflow - OK while computing hash */
+    auto h = utils::hashCode(args.data());
+    switch (h)
+    {
+      case CONST_HASHCODE(MSGPACK):
+        // nobrk
+      case CONST_HASHCODE(MsgPack):
+        // nobrk
+      case CONST_HASHCODE(MessagePack):
+        evtFmt = ETWProvider::EventFormat::ETW_MSGPACK;
+        break;
+
+      case CONST_HASHCODE(XML):
+        // nobrk
+      case CONST_HASHCODE(xml):
+        evtFmt = ETWProvider::EventFormat::ETW_XML;
+        break;
+
+      case CONST_HASHCODE(TLD):
+        // nobrk
+      case CONST_HASHCODE(tld):
+        // nobrk
+        evtFmt = ETWProvider::EventFormat::ETW_MANIFEST;
+        break;
+
+      default:
+        break;
+    }
+#pragma warning(pop)
+    return nostd::shared_ptr<trace::Tracer>{new (std::nothrow) Tracer(*this, name, evtFmt)};
+  }
+};
+}  // namespace ETW
+
+namespace sdk
+{
+namespace trace
+{
+
+class ETWSpanData final : public Recordable
```

Comment:
Good to move this in separate class under `exporters/etw/` directory.

Reply:
Agreed.

Reply:
moved!